File: Java_ChildrenCompetition/src/main/java/service/ITestService.java
package service;

import model.TestDTO;

import java.util.List;

//public interface ITestService  <ID, T extends HasId<ID>>{
//    List<TestDTO> findAllTestDTOs();
//}

public interface ITestService {
    List<TestDTO> findAllTestDTOs();
}

File: CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/service/ITestService.cs
using System.Collections.Generic;
using CSharp_ChildrenCompetitionGUI.model;

namespace CSharp_ChildrenCompetitionGUI.service
{
    public interface ITestService<ID, T> where T : HasId<ID>
    {
        List<TestDTO> findAllTestDTOs();
    }
}